File: algorithms/python/ReverseLinkedListII/reverseBetween.py
def reverseBetween(self, head, m, n):
    if not head or n == m: return head
    p = dummy = ListNode(0)
    dummy.next = head
    for _ in range(m - 1):
        p = p.next
    tail = p.next
    
    for _ in range(n - m):
        temp = p.next
        p.next = tail.next
        tail.next = tail.next.next
        p.next.next = temp
    return dummy.next

File: algorithms/golang/ReverseLinkedListII/reverseBetween.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func reverseBetween(head *ListNode, m int, n int) *ListNode {
    var dummy *ListNode = &ListNode{0, nil}
    dummy.Next = head
    pre := dummy
    for i := 1; i < m; i++ {
        pre = pre.Next
    }
    cur := pre.Next
    for i := m; i < n; i++ {
        nxt := cur.Next
        cur.Next = nxt.Next
        nxt.Next = pre.Next
        pre.Next = nxt
    }
    return dummy.Next
}